api/types/lock.go

```diff
@@ -229,11 +229,11 @@ func (t LockTarget) Match(lock Lock) bool {
 	return (t.User == "" || lockTarget.User == t.User) &&
 		(t.Role == "" || lockTarget.Role == t.Role) &&
 		(t.Login == "" || lockTarget.Login == t.Login) &&
-		(t.Node == "" || lockTarget.Node == t.Node) &&
 		(t.MFADevice == "" || lockTarget.MFADevice == t.MFADevice) &&
 		(t.WindowsDesktop == "" || lockTarget.WindowsDesktop == t.WindowsDesktop) &&
 		(t.AccessRequest == "" || lockTarget.AccessRequest == t.AccessRequest) &&
-		(t.Device == "" || lockTarget.Device == t.Device)
+		(t.Device == "" || lockTarget.Device == t.Device) &&
+		(t.Node == "" && t.ServerID == "" || lockTarget.Node == t.Node || lockTarget.ServerID == t.ServerID)
```

Comment:
Please add parentheses, anyone that can remember the order of precedence between `&&` and `||` has either just looked it up or is lying.

Should we also check `lockTarget.Node == t.ServerID`? Or something where if `t.ServerID` is nonempty we don't actually rely on `t.Node` for anything?

Reply:
7a6bfee379265e0e0eebb2abc1c28b91c897e033 simplifies that and removes internal references to `Node`.

It adds the following check:

```go
		// If the t.Node is non-empty, then the lock target's node or server ID must match it.
		(t.Node == "" || (lockTarget.Node == t.Node || lockTarget.ServerID == t.Node)) &&
		// If the t.ServerID is non-empty, then the lock target's server ID or node must match it.
		// `lockTarget.Node == t.ServerID`` and the previous condition `lockTarget.ServerID == t.Node`
		// complete each other and are both required for backward compatibility with
		// previous versions of Teleport where locking a node was encouraged by setting the Node field.
		(t.ServerID == "" || (lockTarget.ServerID == t.ServerID || lockTarget.Node == t.ServerID))

```